src/mlpack/methods/kde/kde_main.cpp

```diff
@@ -0,0 +1,165 @@
+/**
+ * @file kde_main.cpp
+ * @author Roberto Hueso (robertohueso96@gmail.com)
+ *
+ * Executable for running Kernel Density Estimation.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core/tree/statistic.hpp>
+
+#include "kde.hpp"
+#include "kde_model.hpp"
+
+using namespace mlpack;
+using namespace mlpack::kde;
+using namespace mlpack::util;
+using namespace std;
+
+// Define parameters for the executable.
+PROGRAM_INFO("Kernel Density Estimation",
+    "This program performs a Kernel Density Estimation. KDE is a "
+    "non-parametric way of estimating probability density function. "
+    "For each query point the program will estimate its probability density "
+    "by applying a kernel function to each reference point. Computational "
+    " complexity is O(n^2) but it is optimized by making use of dual-trees. "
```

Comment:
If you want to point this complexity issue out I might suggest this clarification: `The computational complexity of this is O(N^2) where there are N query points and N reference points, but this implementation will typically see better performance as it uses an approximate dual-tree algorithm for acceleration.`